src/server/pkg/storage/chunk/metadata.go

```diff
@@ -0,0 +1,89 @@
+package chunk
+
+import (
+	"context"
+	"crypto/sha512"
+	"encoding/hex"
+
+	"github.com/jmoiron/sqlx"
+)
+
+type ChunkID []byte
+
+func Hash(data []byte) ChunkID {
+	h := sha512.New()
+	h.Write(data)
+	return h.Sum(nil)[:32]
+}
+
+func ChunkIDFromHex(h string) (ChunkID, error) {
+	return hex.DecodeString(h)
+}
+
+func (id ChunkID) HexString() string {
+	return hex.EncodeToString(id)
+}
+
+type ChunkMetadata struct {
+	Size     int
+	PointsTo []ChunkID
+}
+
+type MetadataStore interface {
```

Comment:
I kind of think we should be more consistent between the fileset layer and chunk layer in terms of the store naming convention. Right now we have a notion of a fileset store which will store metadata about a fileset including its root index, and a chunk metadata store which stores metadata about chunks. I think it might make sense to call the fileset store a metadata store as well. Another change related to this is that it is not idiomatic Go to have getters prefixed with get, and I think the `Metadata` suffix is a bit redundant considering the type has metadata in the name. So, function names like `SetChunk`, `Chunk`, `DeleteChunk` seem better to me. Also, it might even be clean to change `ChunkMetadata` -> `Metadata` since it is implicit through the package name what the metadata is associated with. Also, I think the fileset table should be called filesets rather than paths.

Reply:
In the chunk package we should keep the name `MetadataStore`. Rename the methods to `Set`, `Get`, and `Delete`.  I agree with `ChunkMetadata` to `Metadata`.

I'm less sure about the fileset package.  There is only one store for the filesets, the information isn't really spread between 2 stores, the way chunks are. (object storage, and metadata in postgres)

Reply:
Okay, I guess I am fine with the Postgres stores being `fileset.Store` and `chunk.MetadataStore`.